backend/src/test/kotlin/org/loculus/backend/controller/files/RequestUploadEndpointTest.kt

```diff
@@ -0,0 +1,102 @@
+import org.loculus.backend.controller.jwtForAlternativeUser
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+@EndpointTest(
+    properties = ["${BackendSpringProperty.BACKEND_CONFIG_PATH}=$S3_CONFIG"],
+)
+class RequestUploadEndpointTest(
+    @Autowired private val client: FilesClient,
+    @Autowired private val groupManagementClient: GroupManagementControllerClient,
+    @Autowired private val objectMapper: ObjectMapper,
+) {
+
+    @Test
+    fun `GIVEN a request for three URLs THEN returns a response with three URLs`() {
+        val groupId = groupManagementClient.createNewGroup().andGetGroupId()
+        val responseContent = client.requestUploads(groupId, 3)
+            .andExpect(status().isOk)
+            .andReturn()
+            .response
+            .contentAsString
+        val responseJson = objectMapper.readTree(responseContent)
+        assertThat(responseJson.size(), `is`(3))
+        responseJson.forEach {
+            assert(it.has("fileId"))
+            assert(it.has("url"))
+        }
+    }
+
+    @Test
+    fun `GIVEN a request with no numberFiles THEN returns a response with one URL`() {
+        val groupId = groupManagementClient.createNewGroup().andGetGroupId()
+        val responseContent = client.requestUploads(groupId, null)
+            .andExpect(status().isOk)
+            .andReturn()
+            .response
+            .contentAsString
+        val responseJson = objectMapper.readTree(responseContent)
+        assertThat(responseJson.size(), `is`(1))
+        responseJson.forEach {
+            assert(it.has("fileId"))
+            assert(it.has("url"))
+        }
+    }
+
+    @Test
+    fun `GIVEN a request for a URL THEN returns a valid presigned URL`() {
+        val groupId = groupManagementClient.createNewGroup().andGetGroupId()
+        val responseContent = client.requestUploads(groupId, 1)
+            .andExpect(status().isOk)
+            .andReturn()
+            .response
+            .contentAsString
+        val url = objectMapper.readTree(responseContent)
+            .get(0)
+            .get("url")
+            .textValue()
+
+        val content = "test content".toByteArray()
+        val request = HttpPut(url)
+        request.entity = ByteArrayEntity(content, ContentType.TEXT_PLAIN)
+        val httpClient = HttpClients.createDefault()
+        val response = httpClient.execute(request)
+        Assertions.assertEquals(200, response.statusLine.statusCode)
+    }
+
+    @Test
+    fun `GIVEN a request with no groupId THEN returns bad request`() {
+        client.requestUploads(groupId = null, numberFiles = 1)
+            .andExpect(status().isBadRequest)
+    }
+
+    @Test
+    fun `GIVEN a request with groupId of a non-existent group THEN returns not found`() {
+        val groupId = groupManagementClient.createNewGroup().andGetGroupId() + 100
```

Comment:
Maybe test that the group really doesn't exist, this assumes groupId + 100 doesn't exist but who knows

Reply:
I think that we are quite safe here because the database is wiped before each test. – At least, it should be, which reminds me that we forgot to add the new files table: https://github.com/loculus-project/loculus/pull/3978/files#r2063476852

Reply:
agree, I've added the truncation now, we should be fine.